src/Cake.Common/Tools/NuGet/Pack/NuGetPacker.cs

```diff
@@ -175,6 +176,18 @@ private ProcessArgumentBuilder GetArguments(FilePath filePath, NuGetPackSettings
                 builder.Append(settings.Verbosity.Value.ToString().ToLowerInvariant());
             }
 
+            // Properties
+            if (settings.Properties != null && settings.Properties.Count > 0)
+            {
+                if (settings.Properties.Values.Any(x => x == null))
```

Comment:
Does nuget allow empty/white space property values/keys? If not we should probably adjust to 
```cake
if (settings.Properties.Values.Any(string.IsNullOrWhiteSpace))
{
    throw new CakeException("Properties values can not be null or empty.");
}

if (settings.Properties.Keys.Any(string.IsNullOrWhiteSpace))
{
    throw new CakeException("Properties keys can not be null or empty.");
}
```

Strings are evil :wink:  I.e. this is a valid dictionary
```cake
var settings = new {
    Properties = new Dictionary<string, string>{
        {"", ""},
        {" ", " "},
        {"  ", "  "}
    }
};
```